main/src/ca/uwaterloo/flix/language/phase/Reducer.scala

```diff
@@ -178,103 +193,18 @@ object Reducer {
     *
     * We use a concurrent (non-blocking) linked queue to ensure thread-safety.
     */
-  private case class SharedContext(anonClasses: ConcurrentLinkedQueue[AnonClass])
+  private case class SharedContext(anonClasses: ConcurrentLinkedQueue[AnonClass], defTypes: ConcurrentHashMap.KeySetView[MonoType, java.lang.Boolean])
```

Comment:
I am probably getting old and cranky, but what the hell is KeySetView and why do I need it? Can we not just operate on the concurrent hash map? :)

Reply:
Yes, but it would just require you to replace all `x.add(y)` with `x.insert(y, true)` or whatever.

Reply:
This takes the Reducer phase from ~30ms to ~50ms on my machine.
Do you still want it?

Reply:
Is this with XPerf and n=21?

Reply:
This is with `--Xbenchmark-phases`.

Reply:
I want to keep the compiler consistent. So we should do everything the same.

Maybe KeySetView is the right thing to do, but then let us explore it in a separate PR and use it everywhere.

Meanwhile for this PR let us keep it simple stupid, i.e. use a ConcurrentMap with Unit